packages/internals/src/engine-commands/validate.ts

```diff
@@ -0,0 +1,114 @@
+import Debug from '@prisma/debug'
+import chalk from 'chalk'
+import * as E from 'fp-ts/Either'
+import { pipe } from 'fp-ts/lib/function'
+import { match } from 'ts-pattern'
+
+import { ErrorArea, getWasmError, isWasmPanic, RustPanic, WasmPanic } from '../panic'
+import { prismaFmt } from '../wasm'
+import { addVersionDetailsToErrorMessage } from './errorHelpers'
+import { createDebugErrorType, parseQueryEngineError, QueryEngineErrorInit } from './queryEngineCommons'
+
+const debug = Debug('prisma:validate')
+
+export type ValidateOptions = {
+  datamodel: string
+}
+
+export class ValidateError extends Error {
+  constructor(params: QueryEngineErrorInit) {
+    const constructedErrorMessage = match(params)
+      .with({ _tag: 'parsed' }, ({ errorCode, message, reason }) => {
+        const errorCodeMessage = errorCode ? `Error code: ${errorCode}` : ''
+        return `${reason}
+${errorCodeMessage}
+${message}`
+      })
+      .with({ _tag: 'unparsed' }, ({ message, reason }) => {
```

Comment:
In which case does this happen?

Reply:
When the error message (which we assume being a `JSON`-encoded string) isn't parseable via `JSON.parse`.
See [here](https://github.com/prisma/prisma/blob/bfbd3ec077596e4554f56ac20a2cde663cef8b3b/packages/internals/src/engine-commands/queryEngineCommons.ts#L131-L134).